### PR TITLE
[Bugfix:System] Stricter Course Rejoin Requirements

### PR DESCRIPTION
**Pull request by NiaHeermance**
### What is the current behavior?
Students can "rejoin" courses they are already in if they specifically go to the null access page. They can also rejoin archived courses.

### What is the new behavior?
Students can only rejoin in the null section, and the course must be not archived.

### CODE REVIEW

**Review by taorif25 — APPROVED** [review]
Minimal changes, everything seems to be working as intended. Good comments!

**Comment by bmcutler** [comment]
@NiaHeermance 
Please check back on this PR, it failed a Cypress test after I merged in @IDzyre s fix for Cypress testing.  
So it might be a real failure?
Trying another merge master & re-run.

I approve the new logic you've added, but I wonder if its possible that the access information might be flawed (and never not be true).  If a student attempts to access a page (and fails because they've been dropped) do we update the database with the attempt and use that new access data to check for 'recent activity'?  Basically I'm asking if we've tested if its possible to actually fail the access time check?  It can wait for another PR if there is a problem.  I will merge this when it passes the tests.

**Pull request merged**
